src/main/java/javaslang/collection/TreeSet.java

```diff
@@ -92,13 +92,14 @@
 
     public static <T extends Comparable<? super T>> TreeSet<T> ofAll(java.lang.Iterable<? extends T> values) {
         Objects.requireNonNull(values, "values is null");
-        return new TreeSet<>(RedBlackTree.ofAll(values));
+        return values.iterator().hasNext() ? new TreeSet<>(RedBlackTree.ofAll(values)) : empty();
```

Comment:
mmm... I don't understand you :( We have <code>java.lang.Iterable</code> here as parameter.